src/indexes/vector_base.cc

```diff
@@ -353,30 +359,38 @@ absl::StatusOr<uint64_t> VectorBase::TrackKey(const InternedStringPtr &key,
   auto id = inc_id_++;
   auto [_, succ] = tracked_metadata_by_key_.insert(
       {key, {.internal_id = id, .magnitude = magnitude}});
-  TrackVector(id, vector);
+
   if (!succ) {
     return absl::InvalidArgumentError(
         absl::StrCat("Embedding id already exists: ", key->Str()));
   }
+  TrackVector(id, vector);
   key_by_internal_id_.insert({id, key});
   return id;
 }
 
-absl::Status VectorBase::UpdateMetadata(const InternedStringPtr &key,
-                                        float magnitude,
-                                        const InternedStringPtr &vector) {
+absl::StatusOr<bool> VectorBase::UpdateMetadata(
```

Comment:
For a complex return type like this, can we add a comment explaining what it means? Guessing it is error == failure, true == updated, false == not updated? If so, would be good to document under which situation is it not updated